code/game/machinery/shieldgen.dm

```diff
@@ -11,7 +11,7 @@
 		var/health = max_health //The shield can only take so much beating (prevents perma-prisons)
 
 /obj/machinery/shield/New()
-	src.dir = pick(1,2,3,4)
+	dir = pick(1,2,3,4)

```

Comment:
Are we sure our guy understood how directions work...?
Just saying.

Reply:
Clarify please.

Reply:
`dir` is a bitflag

Reply:
I don't think it is. /obj/machinery/shield uses the 'shield-old' sprite from icons/effects/effects.dmi - this sprite does not even have per-direction variants, so regardless of its type it probably does not matter. I thought about taking it out but decided to leave it (without the src) just in case there is some arcane purpose to this dir-setting that I did not understand.

Reply:
These should use the macros anyway. `dir = pick(NORTH, SOUTH, EAST, WEST)`